uvicorn/protocols/http/httptools_impl.py

```diff
@@ -43,9 +43,6 @@ class HttpToolsProtocol(asyncio.Protocol):
     def __init__(
         self, config, server_state, on_connection_lost: Callable = None, _loop=None
     ):
-        if not config.loaded:
-            config.load()
-
```

Comment:
Could you explain why config loading changes are needed? I'd think all we need is to add a flag, or possibly not even that (since it is easy to inspect a function and check if it is an async generator context manager)

Reply:
This is because config's app can't be loaded without a context manager and so it's always loaded by the time it reaches one of these classes